packages/storybook-builder-vite/package.json

```diff
@@ -26,6 +26,7 @@
         "vue-docgen-api": "^4.40.0"
     },
     "peerDependencies": {
+        "@storybook/core-common": ">=6.4.0",
```

Comment:
Wondering if this should be 6.3 instead. Depends on the version of storybook we support I guess.
cc @eirslett ?

Reply:
It's good form to support as wide a version range as possible. However, if it introduces any complexity to support 6.3, I would suggest dropping compatibility for your own sanity's sake. 6.4 is a pretty big change architecturally and most of the community will upgrade within the next few months anyway.

Reply:
I suggest going for 6.4. The Vite builder is quite forward-leaning on dependency versions, it's a similar story with the Vite version we use - if it's more work to support a wider range of versions, we should rather spend time fixing bugs and trying to reach for feature parity with the Webpack builders.

Reply:
Happy to stick with 6.4. Not sure if it is more work though in our current state? We already use 6.3+ versions of storybook packages so adding a peerDependency for 6.4+ seemed a bit weird.

Reply:
Thank you for reviewing so much.

`normalizeStories` is needed to have more control over auto-title-generation, introduced in Storybook 6.4.

cf. https://storybook.js.org/blog/component-story-format-3-0/

Therefore, `normalizeStories` is not necessary for users of version 6.3 who only specify an array of strings for `stories`.

Removed `@storybook/core-common` from peer dependencies.
And now use `normalizeStories` only when it is available, and use simple logic to match types when it is not.
Both version 6.3 and 6.4 will work.